### PR TITLE
Add examples/tests for the ws2812-spi PR  improving the fifo handling

### PR DESCRIPTION
**Pull request by david-sawatzke**
(https://github.com/smart-leds-rs/ws2812-spi-rs/pull/12)
Also contains an example for https://github.com/smart-leds-rs/ws2812-spi-rs/pull/14

### CODE REVIEW

File: avr-examples/examples/avr_ws2812_blink_spi_pre.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    // Create SPI interface.
+    let (spi, _) = spi::Spi::new(
+        dp.SPI,
+        pins.sck.into_output(&mut pins.ddr),
+        pins.mosi.into_output(&mut pins.ddr),
+        pins.miso.into_pull_up_input(&mut pins.ddr),
+        pins.led_rx.into_output(&mut pins.ddr),
+        spi::Settings {
+            clock: spi::SerialClockRate::OscfOver8,
+            ..core::default::Default::default()
```
Comment (Rahix) [inline avr-examples/examples/avr_ws2812_blink_spi_pre.rs]: Does this not work?

```suggestion
            .. Default::default()
```

File: avr-examples/examples/avr_ws2812_blink_spi_pre.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+use ws2812_spi as ws2812;
+
+use crate::ws2812::prerendered::Ws2812;
+use smart_leds::{SmartLedsWrite, RGB8};
+
+#[arduino_leonardo::entry]
+fn main() -> ! {
+    let dp = arduino_leonardo::Peripherals::take().unwrap();
+    let mut pins = arduino_leonardo::Pins::new(dp.PORTB, dp.PORTC, dp.PORTD, dp.PORTE, dp.PORTF);
+    let mut delay = arduino_leonardo::Delay::new();
```
Comment (Rahix) [inline avr-examples/examples/avr_ws2812_blink_spi_pre.rs]: Using `Delay` is more meant for use with `embedded-hal` drivers.  In application code, it's easier to just go with [`arduino_leonardo::delay_ms()`](https://rahix.github.io/avr-hal/arduino_leonardo/fn.delay_ms.html) (Also see below).

File: avr-examples/examples/avr_ws2812_blink_spi_pre.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            g: 0x10,
+            b: 0,
+        };
+        data[2] = RGB8 {
+            r: 0x10,
+            g: 0,
+            b: 0,
+        };
+        ws.write(data.iter().cloned()).unwrap();
+        delay.delay_ms(1000 as u16);
```
Comment (Rahix) [inline avr-examples/examples/avr_ws2812_blink_spi_pre.rs]:
```suggestion
        arduino_leonardo::delay_ms(1000);
```

File: avr-examples/examples/avr_ws2812_blink_spi_pre.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        ws.write(empty.iter().cloned()).unwrap();
+        delay.delay_ms(1000 as u16);
```
Comment (Rahix) [inline avr-examples/examples/avr_ws2812_blink_spi_pre.rs]:
```suggestion
        arduino_leonardo::delay_ms(1000);
```

**Review by Rahix — COMMENTED** [review]
I can only really comment on the `AVR` example right now.  Really cool that you added it, I'll be pointing lot's of people there! Thanks again!

**Comment by david-sawatzke** [comment]
Thanks for the review @rahix! The AVR example works, but requires way more RAM & time than necessary. A ASM solution like the one in FastLED would be way better, but this should work fine for the meantime and be good enough for most applications.

**Pull request merged**